src/common/format.ts

```diff
@@ -67,18 +77,25 @@ export interface InterfaceToken<T = any> {
     prototype: any;
     identity: symbol;
 }
+export interface AliasToken<T = any> {
+    name: string;
+    identity: symbol;
+}
+
 
 export type RtType = RtIntrinsicType | RtObjectType | RtUnionType | RtIntersectionType | RtTupleType | RtArrayType
     | RtGenericType | RtMappedType | RtEnumType | RtCallSite | { TΦ: typeof T_STAND_IN } | RtFunctionType
-    | Function | Literal | InterfaceToken;
+    | Function | Literal | InterfaceToken | AliasToken | RtVariableType | RtAliasType;
 
 export type RtBrandedType = {
     TΦ:
     typeof T_UNION | typeof T_INTERSECTION | typeof T_ANY | typeof T_UNKNOWN | typeof T_VOID | typeof T_UNDEFINED
     | typeof T_NULL | typeof T_TUPLE | typeof T_ARRAY | typeof T_THIS | typeof T_GENERIC | typeof T_MAPPED
-    | typeof T_TRUE | typeof T_FALSE | typeof T_CALLSITE | typeof T_ENUM | typeof T_STAND_IN;
+    | typeof T_TRUE | typeof T_FALSE | typeof T_CALLSITE | typeof T_ENUM | typeof T_STAND_IN | typeof T_ALIAS;
 };
 
+export type RtTypeRef = RtType | (()=>RtType);
```

Comment:
This is ambiguous during reflection, because `RtType` can also be a constructor (ie `class`) which would also have `typeof === 'function'`. Some tricks can be done to determine if its a lambda when inspecting it, but I think they'll cause complication down the road as we extend the type model.

I do see that you've added an isConstructorFunction() test which is one I haven't seen before, unfortunately it can only differentiate between lambdas and normal functions, not whether the function is itself a constructor, because unfortunately all non-lambda functions can be constructors. [See here](https://typescript-rtti.org/#fFYyfHsic291cmNlIjoiY29uc3QgaGFuZGxlciA9IHtcbiAgICBjb25zdHJ1Y3QoKSB7XG4gICAgICAgIHJldHVybiBoYW5kbGVyXG4gICAgfVxufVxuXG5leHBvcnQgY29uc3QgaXNDb25zdHJ1Y3RvckZ1bmN0aW9uID0geCA9PiB7XG4gICAgaWYgKHR5cGVvZiB4ICE9PSAnZnVuY3Rpb24nKSB7XG4gICAgICAgIHJldHVybiBmYWxzZTtcbiAgICB9XG4gICAgdHJ5IHtcbiAgICAgICAgcmV0dXJuICEhKG5ldyAobmV3IFByb3h5KHgsIGhhbmRsZXIpKSgpKVxuICAgIH0gY2F0Y2ggKGUpIHtcbiAgICAgICAgcmV0dXJuIGZhbHNlXG4gICAgfVxufVxuXG5jb25zb2xlLmxvZyhpc0NvbnN0cnVjdG9yRnVuY3Rpb24oKCkgPT4ge30pKTtcbmNvbnNvbGUubG9nKGlzQ29uc3RydWN0b3JGdW5jdGlvbih7fSkpO1xuY29uc29sZS5sb2coaXNDb25zdHJ1Y3RvckZ1bmN0aW9uKGZ1bmN0aW9uKCkgeyB9KSk7IiwiY29tcGlsZXJPcHRpb25zIjoiIn0=) for an example of this.

Even if it did work 100% of the time, it introduces unnecessary complications and confusion throughout the "reflection library" part of the stack (which in the future might not be _our_ reflection library).

Reply:
Right, i will change everything to 
```
()=>RtType
```
So we don't need to use the  isConstructorFunction